pkg/rid/store/cockroach/identification_service_area.go

```diff
@@ -222,3 +222,25 @@ func (c *isaRepo) SearchISAs(ctx context.Context, cells s2.CellUnion, earliest *
 
 	return c.process(ctx, isasInCellsQuery, earliest, latest, pq.Int64Array(cids))
 }
+
+// ListExpiredISAs lists all expired ISAs based on expiredTime and writer
+// expiredTime is compared with ISA.endTime
+func (c *isaRepo) ListExpiredISAs(ctx context.Context, writer string, expiredTime *time.Time) ([]*ridmodels.IdentificationServiceArea, error) {
+	var (
+		isasInCellsQuery = fmt.Sprintf(`
+	SELECT
+		%s
+	FROM
+		identification_service_areas
+	WHERE
+		ends_at > $1
+	AND
+		writer = $2`, isaFields)
+	)
+
+	if len(writer) == 0 {
+		return nil, stacktrace.NewError("Writer is required")
+	}
+
+	return c.process(ctx, isasInCellsQuery, expiredTime, writer)
```

Comment:
ok I think I might see a problem here, if you passed in `expiredTime` we run the risk of using in consistent time from different servers. I understand normally we all use unix epoch but we probably don't want to risk it. We should probably just query it with the SQL statement for getting the database time like `CURRENT_TIME` and do offset with it.